logger/util.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  util.swift
+//  Test
+//
+//  Created by Daniel Brain on 10/11/20.
+//
+
+import Foundation
+
+typealias EmptyCallback = () -> ()
+typealias ResultCallback<T> = (Result<T, Error>) -> ()
+typealias HandlerWithResultCallback<T> = (@escaping ResultCallback<T>) -> ()
+
+func noopEmptyCallback() {}
+func noopResultCallback<T>(_: Result<T, Error>) {}
+
+func debounceWithCallback<T>(time : Double, handler: @escaping HandlerWithResultCallback<T>) -> HandlerWithResultCallback<T> {
+    var callbacks: [ResultCallback<T>] = []
+    var timer: Timer?
+    
+    return { callback in
+        callbacks.append(callback)
+        
+        if let currentTimer = timer {
+            currentTimer.invalidate()
+        }
+        
+        timer = Timer.scheduledTimer(withTimeInterval: time, repeats: false) { _ in
+            handler() { result in
+                callbacks.forEach { innerCallback in
+                    innerCallback(result)
+                }
+            }
+        }
+    }
+}
+
+enum Method: String {
+    case GET, POST
+}
+
+let CONTENT_TYPE = "Content-Type"
+let DEFAULT_CONTENT_TYPE = "application/json"
+let DEFAULT_HEADERS: Headers = [:]
+
+enum RequestError: Error {
+    case NoResponse
+}
```

Comment:
Minor, but I would suggest keeping every `enum` together for improved readability, such as at the top of the file.

Reply:
👍